### ISSUE TITLE
Why the data generator funciton `make_glm_data` for gamma will define n shape parameters for a data set

### ISSUE THREAD

**Issue by belzheng** [abess-team/abess#509]
**Describe the bug**
Why the data generator funciton `make_glm_data` for gamma will define n shape parameters for a data set

```python
elif family == "gamma":
            x = x / 16
            m = 5 * np.sqrt(2 * np.log(p) / n)
            if coef_ is None:
                Tbeta[nonzero] = np.random.uniform(m, 100 * m, k) * sign
            else:
                Tbeta = coef_
            # add noise
            eta = x @ Tbeta + np.random.normal(0, sigma, n)
            # set coef_0 to make eta<0
            eta = eta - np.abs(np.max(eta)) - 10
            eta = -1 / eta
            # set the shape para of gamma uniformly in [0.1,100.1]
            shape_para = 100 * np.random.uniform(0, 1, n) + 0.1
            y = np.random.gamma(
                shape=shape_para,
                scale=eta / shape_para,
                size=n)
```

**Additional context**

Would it be more sensible that a data set share the same shape parameter?

**Comment by Mamba413** [comment]
Thanks for this question. I read "Chapter 8.2 The gamma distribution" in the book "[Generzlized Linear Models](https://www.taylorfrancis.com/books/mono/10.1201/9780203753736/generalized-linear-models-mccullagh)", and find the authors say "we are concerned mostly with models for which the shape parameter is assumed constant for all observations". So I think it is make sense that a data set share the same shape parameter. 

Actually, the simulation setting adopts that in R:
```R
# add noise
sigma <- sqrt((t(beta) %*% Sigma %*% beta) / snr)
eta <- x %*% beta + stats::rnorm(n, 0, sigma)
# set coef_0 as + abs(min(eta)) + 1
eta <- eta + abs(min(eta)) + 10
# set the shape parameter of gamma uniformly in [0.1, 100.1]
shape_para <- 100 * runif(n) + 0.1
y <- stats::rgamma(n, shape = shape_para, rate = shape_para * eta)
```
This is written by @bbayukari , is there has any special consideration?

**Comment by bbayukari** [comment]
It has been fixed in d0394626944649d464493e1c5cb01dcf0db7ccee.
